### DISCUSSION TITLE
Open department links in directory search

### DISCUSSION THREAD

**Discussion by drewvolz** [StoDevX/AAO-React-Native#6430] [Ideas]
They open a url at the moment. This is a search where we could intercept into a new directory search.

**Comment by drewvolz** [comment]
WIP. I've made some progress towards adding this. 

Branch: `drew/directory-department`
Changes: https://github.com/StoDevX/AAO-React-Native/compare/master...drew/directory-department

✅ What is working:
- [x] Tapping on a department fetches department data and renders it natively
- [x] Navigating to a department pushes a new view on to the stack
- [x] Navigating between Query and Department listings works as expected

old | `navigation.navigate`
--|--
 <img alt="old" src="https://user-images.githubusercontent.com/5240843/201222160-eaf3f3ec-eb9c-4616-b744-1a7e6fc84321.gif" />| <img alt="new" src="https://user-images.githubusercontent.com/5240843/201233480-ee87cae1-f560-4d97-abc1-b411306369be.gif" />

**Reply by hawkrives** [reply]
Looking at the gif (I haven't looked at the code yet), it looks like when you tap a department, it's popping the current view and reusing the main search view? I'd expect it to push a _new_ search view onto the stack, and search there?

**Reply by drewvolz** [reply]
I'm calling navigation.navigate at the moment. I would also expect a new view to be added to the stack. Thanks for noting that expectation!

**Reply by drewvolz** [reply]
Updated to `navigation.push` to add this behavior.

Merged in #6571
